### PR TITLE
changed flake.nix

### PR DESCRIPTION
**Pull request by sneedcat**
The overlay wasn't working for some reason, so I repaired it.

### CODE REVIEW

File: flake.lock [hunk 1]
```diff
@@ -49,11 +49,11 @@
...
     "nixpkgs": {
       "locked": {
-        "lastModified": 1641528457,
-        "narHash": "sha256-FyU9E63n1W7Ql4pMnhW2/rO9OftWZ37pLppn/c1aisY=",
+        "lastModified": 1641887635,
+        "narHash": "sha256-kDGpufwzVaiGe5e1sBUBPo9f1YN+nYHJlYqCaVpZTQQ=",
         "owner": "nixos",
         "repo": "nixpkgs",
-        "rev": "ff377a78794d412a35245e05428c8f95fef3951f",
+        "rev": "b2737d4980a17cc2b7d600d7d0b32fd7333aca88",
```
Comment (bbjubjub2494) [inline flake.lock:56]: This is unnecessary to fix the core issue

File: flake.nix [hunk 2]
```diff
@@ -15,41 +15,30 @@
...
+    };
+  } // flake-utils.lib.eachSystem [ "x86_64-linux" "aarch64-linux"] (system:
+    let pkgs = import nixpkgs {
+      inherit system;
+    };
+    in {
+      defaultPackage = self.packages.${system}.polymc;
+      packages = rec {
+        polymc = pkgs.libsForQt5.callPackage ./packages/nix/polymc {
+	  inherit self;
```
Comment (bbjubjub2494) [inline flake.nix]: I don't think it's a good idea to use tabs for indentation.

File: flake.nix [hunk 3]
```diff
@@ -15,41 +15,30 @@
...
+      inherit (self.packages.${final.system})
+      polymc;
+    };
+  } // flake-utils.lib.eachSystem [ "x86_64-linux" "aarch64-linux"] (system:
+    let pkgs = import nixpkgs {
+      inherit system;
+    };
+    in {
+      defaultPackage = self.packages.${system}.polymc;
+      packages = rec {
```
Comment (bbjubjub2494) [inline flake.nix]:
```suggestion
      packages = {
```
redundant rec

File: flake.nix [hunk 4]
```diff
@@ -15,41 +15,30 @@
...
+        polymc = pkgs.libsForQt5.callPackage ./packages/nix/polymc {
+	  inherit self;
+	  submoduleQuazip = quazip;
+	  submoduleNbt = libnbtplusplus;
+	};
+      };
+      apps = rec {
```
Comment (bbjubjub2494) [inline flake.nix]:
```suggestion
      apps = {
```
redundant rec

File: flake.nix [hunk 5]
```diff
@@ -15,41 +15,30 @@
...
+    overlay = final: prev: {
+      inherit (self.packages.${final.system})
+      polymc;
+    };
+  } // flake-utils.lib.eachSystem [ "x86_64-linux" "aarch64-linux"] (system:
+    let pkgs = import nixpkgs {
+      inherit system;
+    };
+    in {
+      defaultPackage = self.packages.${system}.polymc;
```
Comment (bbjubjub2494) [inline flake.nix]:
```suggestion
      defaultPackage = self.packages.${system}.polymc;
      defaultApp = self.apps.${system}.polymc;
```
defaultApp was provided by the previous version.

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]
Hi! You're absolutely right that the overlay has a quirk. I went a bit pedantic on your diff because I think diffs should be held to high standards regardless.

I would recommend picking one of two approaches:
1. If you want to rewrite the `outputs` from nearly scratch, why not? But then I ~~would recommend using `flake-utils.lib.simpleFlake` too keep it tidy.~~ EDIT: nevermind, `simpleFlake` is way too limited.
2. If you want to improve the previous version that's fine too, but then it should be the simplest possible diff that accomplishes that, which this is not.

**Comment by ghost** [comment]
Yooo its the dude from sneedacity

**Comment by sneedcat** [comment]
I made the suggested changes.

I would like to know if you are interested in setting up a [binary cache](https://nixos.wiki/wiki/Binary_Cache). It isn't related to this pull request, but it would be nice for nix users. You can either self host it or use something like cachix for free. The only problem is that you would have to build the package, but that can be automated in a github workflow, [for example](https://github.com/meguminloli/megu-nixpkgs/blob/master/.github/workflows/build.yml).

**Comment by bbjubjub2494** [comment]
Thanks, flake LGTM now. Can you hit `packages/nix/NIX.md` real quick? It mentions `overlay.${system}` once, which is now incorrect.

**Review by bbjubjub2494 — APPROVED** [review]
:+1:

**Review by cidkidnix — APPROVED** [review]

**Pull request merged**